File: java/logbrew-java/src/main/java/co/logbrew/sdk/Transport.java
package co.logbrew.sdk;

/**
 * Public transport interface used by flush and shutdown operations.
 */
@FunctionalInterface
public interface Transport {
    /**
     * Sends a serialized batch body for the configured API key.
     */
    TransportResponse send(String apiKey, String body) throws TransportException;
}

File: php/logbrew-php/src/Transport.php
<?php

declare(strict_types=1);

namespace LogBrew;

/**
 * Public transport contract used by flush and shutdown operations.
 */
interface Transport
{
    /**
     * Send a queued request body through the transport and return its response.
     *
     * @throws TransportError|SdkError
     */
    public function send(string $apiKey, string $body): TransportResponse;
}
